### PR TITLE
Implemented oembed endpoint

### LINKED ISSUE #445: Implement oEmbed endpoint
https://ombed.com
>oEmbed is a format for allowing an embedded representation of a URL on third party sites. The simple API allows a website to display embedded content (such as photos or videos) when a user posts a link to that resource, without having to parse the resource directly.

To support oEmbed, we are going to need to implement an API endpoint at `/v1/oembed`, which will accept parameters `url` and `format`. Fully implement the specification **except for wildcards**. Right now, we only have photos, so only implement the `photo` type. We want to support both JSON and XML formats (this should be as simple as switching Django renderers). Most importantly, we want to add a `license_url` field to the response, which will link to the CC License for an embedded image.

oEmbed requires a `height` and `width` field for images. In some cases, we don't know the dimensions of an image. In those cases, the `oEmbed` endpoint should load the image server-side and measure the dimensions before returning the fully populated response.

The endpoint must be cleanly represented in our generated documentation. There should be some unit tests demonstrating that the endpoint works as expected.

Example request:
GET `https://api.creativecommons.engineering/v1/oembed/?format=json&url=https%3A//search.creativecommons.org/photos/f5af4a72-9ab2-4f5a-a0c5-090e7350c209`

Response:
```
{
  "version": "1.0",
  "type": "photo",
  "width": 1024,
  "height": 768,
  "title": "kittens 022",
  "author_name": "bytehead",
  "author_url": "https://www.flickr.com/photos/66884677@N00",
  "license_url": "https://creativecommons.org/licenses/by-sa/2.0/"
}
```

We would also like to use the `html` field (normally not required for `photo` types) to embed additional rich attribution information as we do in the CC Search front end (optional for contributors; we can add this ourselves).

### PR DESCRIPTION
**Pull request by Tanuj22**
Fixes #445 

A View for Oembed is implemented
---

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by Tanuj22** [comment]
@aldenstpage I am not quite sure if this is the correct approach. Can you please take a look at this?

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 1]
```diff
@@ -261,3 +261,32 @@ def get(self, request, identifier, format=None):
...
+            return Response(data={}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            identifier = url.rsplit('/',1)[1]
+            image_record = Image.objects.get(identifier=identifier)
+        except:
+            return Response(data={}, status=status.HTTP_404_NOT_FOUND)
+        resp = {
+            'version': 1.0,
+            'type': 'photo',
+            'width': width if width else image_record.width,
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: `width` and `height` should always come from the image record, not the query parameters. 

>oEmbed requires a height and width field for images. In some cases, we don't know the dimensions of an image. In those cases, the oEmbed endpoint should load the image server-side and measure the dimensions before returning the fully populated response.

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 2]
```diff
@@ -261,3 +261,32 @@ def get(self, request, identifier, format=None):
...
+            'height': height if height else image_record.height,
+            'title': image_record.title,
+            'author_name': image_record.creator,
+            'author_url': image_record.creator_url,
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: What happens if `creator` or `creator_url` are `None`? Perhaps you can find out through a unit test

File: cccatalog-api/cccatalog/api/views/image_views.py (hunk 1 above)
Comment (Tanuj22) [reply cccatalog-api/cccatalog/api/views/image_views.py]: So should I stop reading `height` and `width` from the url?

File: cccatalog-api/cccatalog/api/views/image_views.py (hunk 2 above)
Comment (Tanuj22) [reply cccatalog-api/cccatalog/api/views/image_views.py]: It would be set as `null`

**Comment by Tanuj22** [comment]
@aldenstpage Requesting review for the current changes. Also I am new to testing, so can you provide me some guidance on how to write tests for the view.

**Comment by Tanuj22** [comment]
@aldenstpage Please have a look at this.

**Comment by Tanuj22** [comment]
@aldenstpage I have written the test. Please have a look

**Pull request merged**